backend/e4h-services/ingestion-service/app/api/endpoints/file_ingestion.py

```
@@ -2437,3 +2440,516 @@ def find_col(partial):
         if input_temp_file and os.path.exists(input_temp_file.name):
             os.unlink(input_temp_file.name)
 
+
+@router.post('/amcConfigurationValidateData',
+             summary='Validate AMC configuration Excel file before processing',
+             response_description='Returns validation report Excel with PASSED/FAILED rows')
+async def validate_amc_configurations_excel_sheet(
+        background_tasks: BackgroundTasks,
+        amc_file: UploadFile = File(..., description="Excel file containing AMC configuration data"),
```

Ruff (0.14.5)

Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)